src/invocation/InvocationService.ts

```diff
@@ -133,13 +140,13 @@ export class InvocationService {
     private invokeSmart(invocation: Invocation) {
         if (invocation.hasOwnProperty('connection')) {
             return this.send(invocation, invocation.connection);
-        } else if (invocation.hasOwnProperty('partitionId')) {
-            var address = this.client.getPartitionService().getAddressForPartition(invocation.partitionId);
+        } else if (invocation.hasPartitionId()) {
+            const address = this.client.getPartitionService().getAddressForPartition(invocation.partitionId);
             return this.sendToAddress(invocation, address);
         } else if (invocation.hasOwnProperty('address')) {
             return this.sendToAddress(invocation, invocation.address);
         } else {
-            return this.send(invocation, this.client.getClusterService().getOwnerConnection());
+            return this.sendToAddress(invocation, this.client.getClusterService().getRandomAddress());
```

Comment:
yep, I see, but during I was finding out what causes an exception in tests I found that in python library client sends to random address, have a look https://github.com/hazelcast/hazelcast-python-client/blob/master/hazelcast/invocation.py#L154

Do I need to return this code back?